src/lib/message_filters.py

```diff
@@ -30,3 +31,22 @@ def filter_time_range(messages, time_key, start_time, end_time):
     @staticmethod
     def filter_noise(messages, message_key, noise_fn):
         return [td for td in messages if not noise_fn(td.get(message_key))]
+    
+    @staticmethod
+    def subsample_messages(messages):
+        '''
+        Generates sample messages 
+        
+        :param: messages: TracedData objects to sample
+        :type traced_data: list of TracedData
+        :return: sample of the TracedData objects
+        :rtype: list of TracedData
+        '''
+
+        subsample_data = []
+        for td in messages:
+            if (td not in subsample_data) and (int(SHAUtils.sha_string(td["uid"])[0], 16)) < 4:
```

Comment:
Do we want the `4` to be in pipeline configuration so we can more easily control this threshold later?

Reply:
yes

Reply:
Yes. Added the threshold to pipeline configuration.